core/nbor/src/de/mod.rs

```diff
@@ -0,0 +1,75 @@
+use std::io::{Cursor, Error, Read};
+use std::mem::size_of;
+
+/// A data-structure that can be de-serialized from binary format by NBOR.
+pub trait Deserializable: Sized {
+    fn read<R: Read>(reader: &mut R) -> Result<Self, Error>;
+
+    /// Deserialize this instance from a slice of bytes.
+    fn from_slice(v: &[u8]) -> Result<Self, Error> {
+        let mut c = Cursor::new(v);
+        Self::read(&mut c)
+    }
+}
+
+macro_rules! impl_for_integer {
+    ($type: ident) => {
+        impl Deserializable for $type {
+            fn read<R: Read>(reader: &mut R) -> Result<Self, Error> {
+                let mut data = [0u8; size_of::<$type>()];
+                reader.read_exact(&mut data)?;
+                Ok($type::from_le_bytes(data))
+            }
+        }
+    };
+}
+
+impl_for_integer!(i8);
+impl_for_integer!(i16);
+impl_for_integer!(i32);
+impl_for_integer!(i64);
+impl_for_integer!(i128);
+impl_for_integer!(isize);
+impl_for_integer!(u8);
+impl_for_integer!(u16);
+impl_for_integer!(u32);
+impl_for_integer!(u64);
+impl_for_integer!(u128);
+impl_for_integer!(usize);
```

Comment:
usize is a bit unclear - as it's different on different platform.
I would actually explicitly not have usize and require cast to u32 or u64 explicitly.